src/python/pants/engine/internals/graph_test.py

```diff
@@ -1012,12 +1010,22 @@ def test_generate_overrides_unused(generated_targets_rule_runner: RuleRunner) ->
         )
 
 
-def test_parametrize(generated_targets_rule_runner: RuleRunner) -> None:
-    assert_generated(
+def assert_parametrized(
+    rule_runner: RuleRunner,
+    build_content: str,
+    addresses: set[Address],
+    expected: set[Target],
+) -> None:
+    rule_runner.write_files({"demo/BUILD": build_content, "demo/f1.ext": "", "demo/f2.ext": ""})
+    result = rule_runner.request(Targets, [Addresses(addresses)])
+    assert set(result) == expected
+
+
+def test_parametrize_tgt_generator(generated_targets_rule_runner: RuleRunner) -> None:
+    assert_parametrized(
         generated_targets_rule_runner,
-        Address("demo"),
         "generator(tags=parametrize(t1=['t1'], t2=['t2']), sources=['f1.ext'])",
-        ["f1.ext"],
+        {Address("demo", parameters={"tags": "t1"}), Address("demo", parameters={"tags": "t2"})},
```

Comment:
Note that we were trying to evaluate the target `demo` - that address is not legal!

Reply:
Also note that none of the expected result changes with these tests, only the input targets change. That's a good thing :)